pandas/core/indexes/base.py

```diff
@@ -1765,6 +1765,29 @@ def _validate_names(
 
         return new_names
 
+    def get_default_index_names(self, names=None, default=None) -> Sequence[str]:
+        from pandas.core.indexes.multi import MultiIndex
+
+        # if names is not None and not all(isinstance(name, str) for name in names):
+        #    raise ValueError("Names must be a string")
+        if names is not None:
+            if isinstance(names, str):
+                names = [names]
+            elif isinstance(names, list) and not all(
```

Comment:
you can use is_list_like

Reply:
why does this have to be a string?

Reply:
My mistake. Condition of index names should only in type of str or list-like.
```
names : str or list-like, default None
``` 
Have make code changes.